src/protocols/secure_channel/CASESession.cpp

```diff
@@ -404,187 +445,283 @@ CHIP_ERROR CASESession::HandleSigmaR1(System::PacketBufferHandle && msg)
+
         VerifyOrExit(mFabricsTable != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
         fabricIndex = mFabricsTable->FindDestinationIDCandidate(ByteSpan(destinationIdentifier), ByteSpan(initiatorRandom),
                                                                 ipkListSpan, GetIPKListEntries());
         VerifyOrExit(fabricIndex != Transport::kUndefinedFabricIndex, err = CHIP_ERROR_CERT_NOT_TRUSTED);
 
         mFabricInfo = mFabricsTable->FindFabricWithIndex(fabricIndex);
         VerifyOrExit(mFabricInfo != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
-    }
 
-    SuccessOrExit(err = tlvReader.Next());
-    VerifyOrExit(TLV::TagNumFromTag(tlvReader.GetTag()) == ++decodeTagIdSeq, err = CHIP_ERROR_INVALID_TLV_TAG);
-    SuccessOrExit(err = tlvReader.GetBytes(mRemotePubKey, static_cast<uint32_t>(mRemotePubKey.Length())));
+        SuccessOrExit(err = tlvReader.Next());
+        VerifyOrExit(TLV::TagNumFromTag(tlvReader.GetTag()) == ++decodeTagIdSeq, err = CHIP_ERROR_INVALID_TLV_TAG);
+        SuccessOrExit(err = tlvReader.GetBytes(mRemotePubKey, static_cast<uint32_t>(mRemotePubKey.Length())));
+
+        SuccessOrExit(err = SendSigma2());
+    }
 
 exit:
 
-    if (err != CHIP_NO_ERROR)
+    if (err == CHIP_ERROR_CERT_NOT_TRUSTED)
+    {
+        SendStatusReport(mExchangeCtxt, kProtocolCodeNoSharedRoot);
+        mState = kInitialized;
+    }
+    else if (err != CHIP_NO_ERROR)
     {
-        SendErrorMsg(SigmaErrorType::kUnexpected);
+        SendStatusReport(mExchangeCtxt, kProtocolCodeInvalidParam);
+        mState = kInitialized;
     }
     return err;
 }
 
-CHIP_ERROR CASESession::SendSigmaR2()
+CHIP_ERROR CASESession::SendSigma2Resume(const ByteSpan & initiatorRandom)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    size_t data_len = EstimateTLVStructOverhead(kCASEResumptionIDSize + kTAGSize + sizeof(uint16_t), 7);
 
-    System::PacketBufferHandle msg_R2;
-    size_t data_len;
+    System::PacketBufferTLVWriter tlvWriter;
+    System::PacketBufferHandle msg_R2_resume;
+    TLV::TLVType outerContainerType = TLV::kTLVType_NotSpecified;
 
-    uint8_t msg_rand[kSigmaParamRandomNumberSize];
+    msg_R2_resume = System::PacketBufferHandle::New(data_len);
+    VerifyOrReturnError(!msg_R2_resume.IsNull(), CHIP_ERROR_NO_MEMORY);
 
-    chip::Platform::ScopedMemoryBuffer<uint8_t> msg_R2_Signed;
-    size_t msg_r2_signed_len;
+    tlvWriter.Init(std::move(msg_R2_resume));
 
-    chip::Platform::ScopedMemoryBuffer<uint8_t> msg_R2_Encrypted;
-    size_t msg_r2_signed_enc_len;
+    // Generate a new resumption ID
+    ReturnErrorOnFailure(DRBG_get_bytes(mResumptionId, sizeof(mResumptionId)));
 
-    uint8_t msg_salt[kIPKSize + kSigmaParamRandomNumberSize + kP256_PublicKey_Length + kSHA256_Hash_Length];
+    ReturnErrorOnFailure(tlvWriter.StartContainer(TLV::AnonymousTag, TLV::kTLVType_Structure, outerContainerType));
+    ReturnErrorOnFailure(tlvWriter.PutBytes(TLV::ContextTag(1), mResumptionId, kCASEResumptionIDSize));
```

Comment:
Odd that in the DRBG call above, `sizeof(mResumptionId)` is used, but here it's `kCASEResumptionIDSize`. Suggest `sizeof(mResumptionId)`